server/dash/dashboards/nc_summary_comparison.py

```diff
@@ -0,0 +1,754 @@
+    html.Div(html.Br(), style=DIVIDER_STYLE),
+    # NC Comparison -  Request Source Bar Charts.
+    html.Div(children=[
+        generate_comparison_req_source_bar("req_source_bar_chart"),
+        generate_comparison_req_source_bar("req_source_bar_chart2")
+    ], style=merge_dict(EQUAL_SPACE_STYLE, {"width": "97.5vw"})),
+    html.Div(html.Br(), style=DIVIDER_STYLE),
+    # NC Comparison - Number of Requests per day Overlapping line chart.
+    generate_comparison_line_chart()
+])
+
+# CALLBACK FUNCTIONS.
+
+
+@callback(
+    [Output("nc_dropdown_filter", "options")],
+    Input("nc_dropdown", "value")
+)
+def generate_dynamic_filter(nc_dropdown):
+    """Enables the dashboard to show dynamic filters.
+
+    This function takes the selected neighborhood council (nc) value from the 
+    "nc_dropdown" dropdown and output a a list of available request types
+    from that neigbhorhood council.
+
+    Args:
+        nc_dropdown: A string argument automatically detected by Dash callback 
+        function when "nc_dropdown" element is selected in the layout.
+
+    Returns: 
+        nc_dropdown_filter: a list of request types available from the selected 
+        neigbhorhood council.
+    """
+    # If no neighborhood council is selected, use all data.
+    if not nc_dropdown:
+        df = api_data_df
+    else:
+        df = api_data_df[api_data_df["councilName"] == nc_dropdown]
+
+    req_types = sorted([n for n in set(df["typeName"])])
+    return req_types
+
+
+def generate_filtered_dataframe(api_data_df, nc_dropdown, nc_dropdown_filter):
+    """Outputs the filtered dataframe based on the selected filters
+
+    This functions takes the original dataframe "api_data_df", selected neighborhood
+     council "nc_dropdown",
+    as well as the selected request type "nc_dropdown_filter" to output a dataframe
+     with matching records.
+
+    Args:
+        api_data_df: full 311-request data directly access from the 311 data API.
+        nc_dropdown: A string argument automatically detected by Dash callback 
+        function when "nc_dropdown" element is selected in the layout.
+        nc_dropdown_filter: A list of strings automatically detected by Dash 
+        callback function when "nc_dropdown_filter" element is selected in the
+         layout, default None.
+    Returns:
+        pandas dataframe filtered by selected neighborhood council and request types
+    """
+    print(" * Generating summary visualizations")
+    if not nc_dropdown:
+        df = api_data_df
+    else:
+        df = api_data_df[api_data_df["councilName"] == nc_dropdown]
+    # Filter as per selection on Request Type Dropdown.
+    if not nc_dropdown_filter and not nc_dropdown:
+        df = api_data_df
+    elif nc_dropdown_filter:
+        df = df[df["typeName"].isin(nc_dropdown_filter)]
+    return df
+
+
+def filter_bad_quality_data(df, data_quality_switch=True):
+    """Filters the dataframe based on pre-defined data quality filters.
+
```

Comment:
It would be nice to note how much of the data this tends to filter out.

Reply:
Thanks for the suggestion - resolved